src/systems/tracked-controls-webvr.js

```diff
@@ -41,26 +41,38 @@ module.exports.System = registerSystem('tracked-controls-webvr', {
    * Update controller list.
    */
   updateControllerList: function () {
-    var controllers = this.controllers;
-    var gamepad;
-    var gamepads;
-    var i;
-    var prevCount;
+    try {
+      var controllers = this.controllers;
+      var gamepad;
+      var gamepads;
+      var i;
+      var prevCount;
 
-    gamepads = navigator.getGamepads && navigator.getGamepads();
-    if (!gamepads) { return; }
+      gamepads = navigator.getGamepads && navigator.getGamepads();
+      if (!gamepads) { return; }
 
-    prevCount = controllers.length;
-    controllers.length = 0;
-    for (i = 0; i < gamepads.length; ++i) {
-      gamepad = gamepads[i];
-      if (gamepad && gamepad.pose) {
-        controllers.push(gamepad);
+      prevCount = controllers.length;
+      controllers.length = 0;
+      for (i = 0; i < gamepads.length; ++i) {
+        gamepad = gamepads[i];
+        if (gamepad && gamepad.pose) {
+          controllers.push(gamepad);
+        }
       }
-    }
 
-    if (controllers.length !== prevCount) {
-      this.el.emit('controllersupdated', undefined, false);
+      if (controllers.length !== prevCount) {
+        this.el.emit('controllersupdated', undefined, false);
+      }
+    } catch (e) {
+      if (e.name === 'SecurityError') {
+        if (window.self === window.top) {
+          console.warn('A-Frame requires additional permissions to list the gamepads. The HTTP `Permissions-Policy` header must not exclude this origin in the `gamepad` directive.', e);
+        } else {
+          console.warn('A-Frame requires additional permissions to list the gamepads. The iframe `allow` attribute must not block this origin.', e);
+        }
+      } else {
+        console.error('Can\'t update controller list:', e);
```

Comment:
why is this error and above just warnings?

Reply:
If the A-frame app doesn't need the controllers, failing to get the controllers will not keep the app from working. If the a-frame app does need controllers, not getting the controllers is a problem that has to be fixed. Since this code can't determine which, a warning is reasonable.  (However, if the error isn't caught at all, A-Frame doesn't fully initialize.)

"Can\'t update controller list" would only occur if there was some unforeseen error and would likely be a programming error in A-Frame.